Automattic/wp-calypso
# Pricing: Make feature comparison and "for agencies" links locale-aware

This PR fixes a bug where even on localized versions of the pricing page, the links leading people to Jetpack's feature comparison page and the "For Agencies" landing page point to the English version of those pages.

If browsing the pricing page in the Jetpack Cloud (i.e., Calypso Green) environment, these links now also preserve the root domain for folks who are testing outside of `cloud.jetpack.com` (e.g., `jetpack.cloud.localhost`). Browsing from WordPress.com should still redirect to the staging/production versions.

Resolves `1202858161751496-as-1204020596371942`.

## Proposed Changes

* Create two new custom hooks to calculate the correct locale-aware links for `cloud.jetpack.com/features/comparison` and `jetpack.com/for/agencies` (using `useLocale`).
* Remove the `PLAN_COMPARISON_PAGE` and `AGENCIES_PAGE` constants, since they're no longer... constant.
* Remove an unused component: `SeeAllFeatures`.

## Testing Instructions

1. Visit `/{locale}/pricing` in Jetpack Cloud, using whichever locale you choose (or empty, for English). Ignore any locale slugs that return a "page not found" error. Test as many as you can; for example:

* `ar`
* `es`
* `fr`
* `he`
* `hu`
* `ja`
* `ko`
* `pt`
* `pt-br`
* `pt`
* `zh`
* `zh-cn`
* `zh-sg`
* `zh-tw`

2. Scroll down to the bottom of the pricing grid.
3. Click on both the "Compare all product bundles" and "Explore Jetpack for Agencies" links. Verify that in both cases you're sent to the correct page for your selected locale. *(NOTE: Some locales may not have their own version of a page, in which case the expected result is the English version.)*
4. In Calypso Blue, visit `/plans/{site}` for a self-hosted Jetpack site. Scroll down to the same links and click them. Verify you're directed to the same locale-aware pages, but on `jetpack.com` or `cloud.jetpack.com` as opposed to any other domain.

<img width="655" alt="image" src="https://user-images.githubusercontent.com/670067/225906705-207d0925-6036-49aa-a402-1b27d7ab7482.png">

## Pre-merge Checklist




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1204020596371942
